Extensions/audio_plus.js

```diff
@@ -346,10 +334,44 @@ XKit.extensions.audio_plus = {
 
 		audio_plus.current_player = player;
 		audio_plus.pop_out_controls.classList.add("showing");
-		document.body.classList.add("xkit_audio_plus_popout_showing");
 		audio_plus.pop_out_controls.classList.add("playing");
+		var ppIcon = audio_plus.pop_out_controls.querySelector('.play-pause').querySelector('.icon');
+		ppIcon.classList.remove("icon_play");
+		ppIcon.classList.add("icon_pause");
+		if (audio_plus.can_see_docked_posts) {
+			$("#right_column").classList.add("has_docked_audio");
+		}
 	},
 
+	progress_observer: new MutationObserver(function(mutations, observer) {
+		for (var mutation of mutations) {
+			XKit.extensions.audio_plus.pop_out_controls_progress.setAttribute("style", mutation.target.attributes.getNamedItem("style").value);
+			//reset when audio is finished
+			if (mutation.target.attributes.getNamedItem("style").value == "width: 0px;") {
+				XKit.extensions.audio_plus.controls_click_callback();
+			}
+		}
+	}),
+
+	icon_observer: new MutationObserver(function(mutations, observer) {
```

Comment:
i don't think this needs a new mutationobserver - there's already logic a few lines above to do something when the track ends, just extend that?

Reply:
Unfortunately the `progress_observer` only checks the progress element's width on the original audio post, it doesn't check for the icon element's classes. I don't think there's a way to check if the icon changed on the original post (by the user pausing/playing the track) using the progress element? But if I missed something please let me know!

Reply:
ah, i didn't consider listening for that at all! nice touch!

Reply:
Thanks! ^__^